tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs

```diff
@@ -172,23 +173,20 @@ private void OnBeginRequest(object sender, EventArgs eventArgs)
                 var security = Security.Instance;
                 if (security.Settings.Enabled)
                 {
-                    security.InstrumentationGateway.RaiseRequestStart(httpContext, httpContext.Request, scope.Span);
+                    SecurityCoordinator.ReportWafInitInfoOnce(security, scope.Span);
+                    var securityCoordinator = new SecurityCoordinator(security, httpContext, scope.Span);
 
-                    if (httpRequest.ContentType?.IndexOf("application/x-www-form-urlencoded", StringComparison.InvariantCultureIgnoreCase) >= 0)
-                    {
-                        var formData = new Dictionary<string, object>();
-                        foreach (string key in httpRequest.Form.Keys)
-                        {
-                            formData.Add(key, httpRequest.Form[key]);
-                        }
+                    // request args
+                    var args = securityCoordinator.GetBasicRequestArgsForWaf();
 
-                        security.InstrumentationGateway.RaiseBodyAvailable(httpContext, scope.Span, formData);
+                    // body args
+                    if (httpRequest.ContentType.IndexOf("application/x-www-form-urlencoded", StringComparison.InvariantCultureIgnoreCase) >= 0)
```

Comment:
I _think_ `ContentType` could be null, so may need a `?`?
```suggestion
                    if (httpRequest.ContentType?.IndexOf("application/x-www-form-urlencoded", StringComparison.InvariantCultureIgnoreCase) >= 0)
```

Reply:
I'm  getting the `condition access qualifier expression is known to be not null` from rider, but I'll add it anyway in case!
The source code is

 ```
public string ContentType
      get
      {
        if (this._contentType == null)
        {
          if (this._wr != null)
            this._contentType = this._wr.GetKnownRequestHeader(12);
          if (this._contentType == null)
            this._contentType = string.Empty;
        }
        return this._contentType;
      }

```